### PR TITLE
4203: Replace view solution with view solution component

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/view-solution/controller.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-export const getPublicSolutionById = async (solutionId) => {
-  const response = await new ApiProvider().getPublicSolutionById(solutionId);
+export const getPublicSolutionById = async ({ solutionId }) => {
+  const response = await new ApiProvider().getPublicSolutionById({ solutionId });
 
   if (response.data) {
     logger.info(`Solution ${solutionId}: ${response.data.name} returned`);
-    return createViewSolutionPageContext(response.data);
+    return createViewSolutionPageContext({ previewData: response.data });
```
Comment (alibombaye) [inline app/pages/view-solution/controller.js]: can we call it something else please... `solutionData` perhaps

File: app/pages/view-solution/template.njk [hunk 2]
```diff
@@ -1,60 +1,20 @@
...
-{% from 'components/view-section-heading/macro.njk' import viewSectionHeading %}
-{% from 'components/tag/macro.njk' import tag %}
+
+{% from 'sections/view-solution/macro.njk' import viewSolution %}
 
 {% block body %}
-  <div data-test-id="view-solution-page" class="nhsuk-width-container nhsuk-u-margin-top-4 nhsuk-u-margin-bottom-9">
-    <div data-test-id="view-solution-page-back-link" class="nhsuk-u-margin-bottom-5">
+  <div class="nhsuk-width-container nhsuk-u-margin-top-5">
+      <div data-test-id="view-solution-page-back-link" class="nhsuk-u-margin-bottom-5">
```
Comment (alibombaye) [inline app/pages/view-solution/template.njk]: 🙄

Comment (kat1906) [reply app/pages/view-solution/template.njk]: lol

Comment (kat1906) [reply app/pages/view-solution/template.njk]: this one is right

Comment (alibombaye) [reply app/pages/view-solution/template.njk]: really? 
Should it not be indented just 2 spaces?

File: app/pages/view-solution/controller.js (hunk 1 above)
Comment (kat1906) [reply app/pages/view-solution/controller.js]: ok

File: app/pages/view-solution/template.njk (hunk 2 above)
Comment (kat1906) [reply app/pages/view-solution/template.njk]: I knew I should have written that nunjucks linter 😄

File: app/pages/view-solution/template.njk [hunk 3]
```diff
@@ -4,11 +4,11 @@
 
 {% block body %}
   <div class="nhsuk-width-container nhsuk-u-margin-top-5">
-      <div data-test-id="view-solution-page-back-link" class="nhsuk-u-margin-bottom-5">
-      {{ backLink({
-        "href": "./",
-        "text": "Go back to previous page"
-      }) }}
+    <div data-test-id="view-solution-page-back-link" class="nhsuk-u-margin-bottom-5">
+    {{ backLink({
```
Comment (alibombaye) [inline app/pages/view-solution/template.njk]: 😨 should the macro bit inside be indented

File: app/pages/view-solution/solutionPageContext.test.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ describe('createViewSolutionPageContext', () => {
       },
     };
 
-    const context = createViewSolutionPageContext({ previewData: newPreviewData });
+    const context = createViewSolutionPageContext({ solutionData: newPreviewData });
```
Comment (alibombaye) [inline app/pages/view-solution/solutionPageContext.test.js]: 😨 could we change newPreviewData please

File: app/pages/view-solution/template.njk [hunk 5]
```diff
@@ -5,10 +5,10 @@
...
   <div class="nhsuk-width-container nhsuk-u-margin-top-5">
     <div data-test-id="view-solution-page-back-link" class="nhsuk-u-margin-bottom-5">
-    {{ backLink({
-      "href": "./",
-      "text": "Go back to previous page"
-    }) }}
+      {{ backLink({
+        "href": "./",
+        "text": "Go back to previous page"
+      }) }}
```
Comment (alibombaye) [inline app/pages/view-solution/template.njk]: beautiful 😍

File: app/pages/view-solution/solutionPageContext.test.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  solutionHeader: {
+    id: '100000-001',
+    name: 'Write on Time',
+    supplierName: 'Really Kool Corporation',
+    isFoundation: true,
+    lastUpdated: '1996-03-15T10:00:00',
+  },
+};
+
+const previewData = {
```
Comment (alibombaye) [inline app/pages/view-solution/solutionPageContext.test.js]: mmmm `solutionData` please

File: app/pages/view-solution/solutionPageContext.test.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  id: '100000-001',
+  name: 'Write on Time',
+  supplierName: 'Really Kool Corporation',
+  isFoundation: true,
+  lastUpdated: '1996-03-15T10:00:00',
+};
+
+describe('createViewSolutionPageContext', () => {
+  it('should return the sections provided in the previewData', () => {
```
Comment (alibombaye) [inline app/pages/view-solution/solutionPageContext.test.js]: solutionData

**Pull request merged**